nifi-nar-bundles/nifi-spark-bundle/nifi-livy-processors/src/test/java/org/apache/nifi/processors/livy/ExecuteSparkInteractiveTestBase.java

```diff
@@ -64,33 +66,37 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
                     }
                     session1Requests++;
                 }
-
-                response.setContentLength(responseBody.length());
-
-                try (PrintWriter writer = response.getWriter()) {
-                    writer.print(responseBody);
-                    writer.flush();
-                }
-
             } else if ("POST".equalsIgnoreCase(request.getMethod())) {
-
-                String responseBody = "{}";
-                response.setContentType("application/json");
-
-                if ("/sessions".equalsIgnoreCase(target)) {
-                    responseBody = "{\"id\": 1, \"kind\": \"spark\", \"state\": \"idle\"}";
-                } else if ("/sessions/1/statements".equalsIgnoreCase(target)) {
-                    responseBody = "{\"id\": 7}";
+                String requestBody = IOUtils.toString(request.getReader());
+                try {
+                    System.out.println("requestBody: " + requestBody);
+
+                    new ObjectMapper().readTree(requestBody);
```

Comment:
may you please add some comments explaining what and why you are doing this? It is clear since we are in the context of this PR, but for future readers I think a comment would be very helpful.

Reply:
Done.